### DISCUSSION TITLE
Are there any plans to develop a plugin for RustRover?

### DISCUSSION THREAD

**Discussion by Anivie** [slint-ui/slint#3474] [General]
Jetbrains has announced the launch of RustRover, an IDE designed specifically for Rust. Does slint have any plans to adapt it? At present, it seems that the slint macro is almost unavailable on RustRover.

**Comment by ogoffart** [comment]
It was just announced, and we don't really have plans.
I don't know much about the technical but I hope they support the LSP, then hopefully it works out of the box.

**Comment by 90th** [comment]
looking forward to this plugin if it ever happens.

**Comment by felipetesc** [comment]
It would be nice to allow Slint to work with more IDE(s) actually. Perhaps to create Slintui as an OS process accessible through IPC, could allow Slint to be used with Warp, Zed, VSCode, RustRover, and so many other softwares. Also, the project would have a single way to deal with different applications.  

Off-topic: More and more, people want apps with the possibility to collaborate, if Slint's library process can receive network data to change its inner state the project would so much more. (Should I create an new feature request for this, is the idea even interesting ?)

Thanks

**Reply by ogoffart** [reply]
slint-lsp is exactly that process communicating with IPC, and that means that every editor that supports the LSP protocol should have Slint integration.

We document how to configure some of them there: https://github.com/slint-ui/slint/blob/master/editors/README.md
But this is only for editors that were tried. If you find out how to use slint-lsp with your editor, then please contribute to this page to add instructions for it.

**Reply by felipetesc** [reply]
> slint-lsp is exactly that process communicating with IPC, and that means that every editor that supports the LSP protocol should have Slint integration.
> 
> We document how to configure some of them there: https://github.com/slint-ui/slint/blob/master/editors/README.md But this is only for editors that were tried. If you find out how to use slint-lsp with your editor, then please contribute to this page to add instructions for it.

This is nice! I'll check right away.
